components/reference-implementation-openbanking-nextgenpsd2/src/main/java/org/wso2/openbanking/nextgenpsd2/extensions/handlers/impl/AccountConsentAuthorizeHandler.java

```diff
@@ -0,0 +1,207 @@
+        String userId = requestData.getUserId();
+        switch (ExtensionEnums.PermissionEnum.fromValue(permission)) {
+            case DEDICATED_ACCOUNTS:
+                consentData.setPermissions(AccountConsentUtil.buildPermissionsForDedicatedAccounts(receipt, userId));
+                break;
+            case ALL_PSD2:
+                consentData.setPermissions(AccountConsentUtil.buildPermissionsForAllPsd2Accounts(userId));
+                break;
+            case AVAILABLE_ACCOUNTS:
+            case AVAILABLE_ACCOUNTS_WITH_BALANCES:
+                consentData.setPermissions(AccountConsentUtil.buildPermissionsForAvailableAccounts(userId, permission));
+                break;
+            case BANK_OFFERED:
+                // Build both consumer data with consumer accounts and consent data with permissions
+                /*
+                ToDo: When an bank offered account consent requires multi-level authorization, how account selection
+                 happens is not specified in the nextGenPSD2 specification. Depending on how this issue is resolved,
+                 the following account selection might need be modified.
+                 */
+                AccountConsentUtil.populatePermissionsAndConsumerAccounts(responseData, receipt, userId);
+                break;
+            default:
+                throw new ExtensionException(Response.Status.BAD_REQUEST, "invalid_request",
+                        "Unidentified permission found.");
+        }
+    }
+
+    /**
+     * Returns authorizations and account mappings for account consent persistence.
+     * This method is responsible for,
+     * <ul>
+     *     <li>amending consent status</li>
+     *     <li>amending authorization status of the authorized resource</li>
+     *     <li>adding account permission mappings to the authorized resource</li>
+     *     <li>updating consent receipt in case they were selected by PSU from a list of available accounts</li>
+     * </ul>
+     *
+     * @param requestBody body of the request received by persist-authorized-consent for account consents
+     * @param authorizingResource authorization resource being authorized for account consents
+     * @return amended account consent resource
+     */
+    @Override
+    public DetailedConsentResourceDataWithAmendments getAmendedConsentResource(
+            PersistAuthorizedConsentRequestBody requestBody, StoredAuthorization authorizingResource)
+            throws AuthorizationFailureException, ExtensionException {
+
+        // Build response consent resource
+        PersistAuthorizedConsent requestData = requestBody.getData();
+        boolean isApproved = requestData.getIsApproved();
+
+        // Get auth status from approval
+        String authStatus;
+        if (isApproved) {
+            authStatus = ExtensionEnums.ScaStatusEnum.FINALISED.toString();
+        } else {
+            authStatus = ExtensionEnums.ScaStatusEnum.FAILED.toString();
+        }
```

Comment:
something like below should work

```suggestion
        String authStatus = isApproved ? ExtensionEnums.ScaStatusEnum.FINALISED.toString() ? ExtensionEnums.ScaStatusEnum.FAILED.toString();
```

Reply:
Fixed.